index.c

```diff
@@ -1,118 +1,186 @@
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
-#include <ctype.h>
 
-int od_top = -1, op_top = -1;
-int operands_st[1000], operators_st[1000];
+struct Record {
+    int userID;
+    char userName[50];
+    int userAge;
+};
 
-int isOperand(char ch) {
-    return (ch >= '0' && ch <= '9');
+void ensureFileExists() {
+    FILE *dataFile = fopen("user_records.txt", "a");
+    if (dataFile) fclose(dataFile);
 }
 
-int isOperator(char ch) {
-    return (ch == '+' || ch == '-' || ch == '*' || ch == '/');
+void addRecord() {
+    struct Record record;
+    FILE *dataFile = fopen("user_records.txt", "a");
+
+    if (!dataFile) {
+        printf("Error: Could not access the file.\n");
+        return;
+    }
+
+    printf("Enter ID: ");
+    scanf("%d", &record.userID);
+    getchar();
+    printf("Enter Name: ");
+    fgets(record.userName, sizeof(record.userName), stdin);
+    record.userName[strcspn(record.userName, "\n")] = '\0';
+    printf("Enter Age: ");
+    scanf("%d", &record.userAge);
+
+    fprintf(dataFile, "%d,%s,%d\n", record.userID, record.userName, record.userAge);
```

Comment:
Can we create a method to display the data and reuse it instead of writing the same code repeatedly?